src/sentry/search/events/builder/metrics.py

```diff
@@ -772,6 +774,53 @@ def resolve_limit(self, limit: Optional[int]) -> Optional[Limit]:
     def resolve_granularity(self) -> Granularity:
         return Granularity(self._granularity)
 
+    def _get_snql_using_metrics_layer(self, snuba_request: Request) -> Request:
+        # We use directly the query builder which is not a good practice, considering that the metrics layer
+        # should hide snql generation.
+        from sentry.snuba.metrics import SnubaQueryBuilder
+        from sentry.snuba.metrics.mqb_query_transformer import transform_mqb_query_to_metrics_query
+
+        snuba_queries, _ = SnubaQueryBuilder(
+            projects=self.params.projects,
+            metrics_query=transform_mqb_query_to_metrics_query(
+                snuba_request.query, is_alerts_query=True
+            ),
+            use_case_id=UseCaseKey.PERFORMANCE
+            if self.is_performance
+            else UseCaseKey.RELEASE_HEALTH,
+        ).get_snuba_queries()
+
+        if len(snuba_queries) == 0 or len(snuba_queries) > 1:
+            raise NotImplementedError(
+                "get_snql_query cannot be implemented for MetricsQueryBuilder"
```

Comment:
This is no longer a valid error message in this case

Reply:
You are right, will make it more meaningful!